### PR TITLE
Next

### PR DESCRIPTION
**Pull request by Shrutikabansal**

### CODE REVIEW

**Comment by gmuthukr** [comment]
Please mention what this PR would do in subject (and in description if needed).

File: netcontrold/tests/lib/test_pmd_load_variance.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#
+
+from unittest import mock
+from unittest import TestCase
+
+from netcontrold.app import ncd
+from netcontrold.lib import dataif
+from netcontrold.lib import config
+
+_BASIC_CPU_INFO = """
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_pmd_load_variance.py]: This is not cpu info!.

File: netcontrold/tests/lib/test_pmd_load_variance.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class TestUtil_pmd_load_variance(TestCase):
+
+    @mock.patch('netcontrold.lib.dataif.open')
+    def setUp(self, mock_open):
+        mock_open.side_effect = [
+            mock.mock_open(read_data=_BASIC_CPU_INFO).return_value
+        ]
+        self.pmd_map = dict()
+        self.pmd_map = dataif.get_pmd_stats({})
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_pmd_load_variance.py]: No ack for this approach!...
All you need to do is start creating pmd objects manually as in PR #47 and then call pmd_load_variance.
When you set various cycle data, check how pmd_load is calculated in the application and compare manually calculated value with what returned. Of course, add negative tests as well as proposed in the doc.

Thanks.

**Review by gmuthukr — CHANGES_REQUESTED** [review]
Rework on how pmd objects are created and resubmit with actual tests.

**Comment by Shrutikabansal** [comment]
Here i changed in one test case for pmd_load_variance.

**Comment by Shrutikabansal** [comment]
Changes done in dataif.py file and test file

File: netcontrold/lib/dataif.py [hunk 3]
```diff
@@ -618,6 +618,8 @@ def pmd_load_variance(pmd_map):
     pmd_map : dict
         mapping of pmd id and its Dataif_Pmd object.
     """
+    update_pmd_load(pmd_map)
```
Comment (gmuthukr) [inline netcontrold/lib/dataif.py:622]: The caller has to ensure updating pmd load before calling this function. Discard this change.

File: netcontrold/tests/lib/test_pmd_load_variance.py [hunk 4]
```diff
@@ -13,46 +13,39 @@
...
 import copy
-from unittest import mock
 from unittest import TestCase
-
-from netcontrold.app import ncd
+from unittest import mock
 from netcontrold.lib import dataif
 from netcontrold.lib import config
 
-_BASIC_DATA = """
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_pmd_load_variance.py]: Please squash your commits you work for these test and submit one for test ..

File: netcontrold/tests/lib/test_pmd_load_variance.py [hunk 5]
```diff
@@ -13,46 +13,39 @@
...
+physical id     : 0
+
+processor       : 3
+core id         : 1
+physical id     : 0
+"""
 
-class TestUtil_pmd_load_variance(TestCase):
+
+class TestRebalDryrunRR_OnePmd(TestCase):
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_pmd_load_variance.py:41]: I think something you screwed up pulling old commit that I made .. it shows my additions.... I drop reviewing it and wait for new PR with fresh code changes.
Thanks.

**Review by gmuthukr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**